Make icons clickable to show shipment info

vipin8169 opened the pull request:
Closes #2

vipin8169 committed:
Make icons clickable to show shipment info #2

coderbyheart reviewed (changes requested):
![image](https://user-images.githubusercontent.com/188915/79801388-5a196680-835e-11ea-90e3-f592b79a9bf4.png)

The formatting of the values is not the same as in the summary.
Please re-use the existing logic for formatting numbers in the project.

![image](https://user-images.githubusercontent.com/188915/79801767-03f8f300-835f-11ea-8d1b-ce89ea3013cb.png)

When clicked, the popup hides the marker, please position the popup, so that the marker is visible:

![image](https://user-images.githubusercontent.com/188915/79801838-2559df00-835f-11ea-9709-b4061e6c3176.png)

vipin8169 commented:
I am not sure why it is showing up that way for you. For me it was showing up differently, as shown in screenshot below.
![image](https://user-images.githubusercontent.com/4578304/79801868-c3c05300-8313-11ea-8c4a-50d23f6c7e34.png)

vipin8169 committed:
Used the same number formatting for info as used in summary stats.

Applied an offset of [0,-15] to the pop-up so that it does not hide the marker.

vipin8169 committed:
Some minor code formatting changes

coderbyheart reviewed (changes requested):
![Screenshot_20200421-081621~2](https://user-images.githubusercontent.com/188915/79832271-61199680-83a9-11ea-9204-9fe9639ab3f7.png)


The formatting of the weight is not the same,
please turn the formatting code into a reusable function so it is the same in the summary and the tooltip.

coderbyheart commented on src/Map.tsx:
```diff
@@ -57,23 +68,38 @@ export const Map = ({
-								position={destination.position}
-							/>
+								position={destination.position}>
+								<Popup offset={[0, -15]}>
```
Instead of setting the offset on the Popup, use the `popupAnchor` propery on the Icon: https://leafletjs.com/reference-1.6.0.html#icon-popupanchor

This allows to swap out the icon, without needing to update the Popup definition.

vipin8169 committed:
Used popupAnchor to position the popup instead of applying offest

vipin8169 commented:
> ![Screenshot_20200421-081621~2](https://user-images.githubusercontent.com/188915/79832271-61199680-83a9-11ea-9204-9fe9639ab3f7.png)
> 
> The formatting of the weight is not the same,
> please turn the formatting code into a reusable function so it is the same in the summary and the tooltip.

I am using the same function which was used in summary. it should produce the same output as summary. 
Also, I am not sure where would you like for me to put the reusable code (which file). Please advise which file it should go to for easy reference.

coderbyheart commented:
> I am using the same function which was used in summary. it should produce the same output as summary.

No, you are not. You have made a copy, and the output for weight in the tooltip is in `kg` while the output in the summary is in `t`.

> Also, I am not sure where would you like for me to put the reusable code (which file). Please advise which file it should go to for easy reference.

Ok, here you go:

1. create a file called `src/formatters.ts`
2. move the `numberFormatter` and `currencyFormatter` from [`src/Statistics.tsx`](https://github.com/distributeaid/shipments/blob/saga/src/Statistics.tsx) in there
2. create an arrow function in there called `formatWeight`, which uses the `numberFormatter` and applies `Math.round(weight / 1000)` to return the formatted weight in tons.
3. create an arrow function in there called `formatValue`, which returns the formatted value in Euro
4. Update [`src/Statistics.tsx`](https://github.com/distributeaid/shipments/blob/saga/src/Statistics.tsx) to use the `formatWeight` and `formatValue` functions
5. Update `src/Map.tsx` to use the `formatWeight` and `formatValue` functions

vipin8169 committed:
Reformatted the code to put reusable formatting functions in a separate file

vipin8169 commented:
Thank you for your help :)
I am sorry, I assumed the data to be displayed in kg instead of tonnes, since info in data file was in kgs. I have fixed that, and hopefully addressed all the concerns. Thank you!

coderbyheart commented on src/formatter.ts:
```diff
@@ -0,0 +1,15 @@
+
+export const formatWeight = (weight: number) => {
+    return (numberFormatter.format(Math.round(weight / 1000)) + ' t')
+}
```
```suggestion
export const formatWeight = (weight: number) => `${numberFormatter.format(Math.round(weight / 1000))} t`
```

coderbyheart reviewed (changes requested):
Changes look good, now.

Please format source code with prettier (do not use `--no-verify` when committing!)

vipin8169 committed:
refactor: reformatted the code with prettier

coderbyheart commented:
Great! :+1:
